src/commands/data/schemas/util/syncSchema.ts

```diff
@@ -359,3 +297,185 @@ function deepDiff(object: any, other: any) {
     }
   });
 }
+
+function diffRootAttributes(localSchema: any, cloudSchema: any) {
+  return deepDiff(
+    _.pick(localSchema, 'description', 'defaultLimit', 'maximumLimit', 'createMode', 'readMode', 'updateMode', 'deleteMode', 'groupSyncMode'),
+    _.pick(cloudSchema, 'description', 'defaultLimit', 'maximumLimit', 'createMode', 'readMode', 'updateMode', 'deleteMode', 'groupSyncMode')
+  );
+}
+
+function logDifferentRootAttributes(cloudSchema: any, updatedValues: any) {
+  const changedKeys = Object.keys(updatedValues);
+  console.group(`Schema ${cloudSchema.name} - Root attributes`);
+  if (changedKeys.length < 1) { console.log('No update required.'); }
+  for (const key of changedKeys) {
+    console.log(`${chalk.yellow(key)}:\t ${chalk.red(cloudSchema[key])} => ${chalk.green(updatedValues[key])}`);
+  }
+  console.groupEnd();
+}
+
+function logChanges(group: string, { toAdd, toRemove, toUpdate }: {
```

Comment:
Again just for readability / simplicity

```
type changes = { toAdd: string[]; toRemove: string[]; toUpdate: string[]; };
function reportSchemaChanges(group: string, changes: changes) {
  const { toAdd, toRemove, toUpdate } = changes;

  console.group(group);

  // Should resolve correctly without the > 1 , 0 is falsy right?
  if (toAdd.length && toRemove.length && toUpdate.length) {
    console.log('No update required');
    return;
  }

  // Report changes to the schema
  forEach(toAdd, key => console.log(`Will be added: ${chalk.green(getIdentifier(key))}`));
  forEach(toRemove, key => console.log(`Will be removed: ${chalk.red(getIdentifier(key))}`));
  forEach(toUpdate, key => console.log(`Will be updated: ${chalk.yellow(getIdentifier(key))}`));

  console.groupEnd();
}
```

Reply:
I would also take this example and apply the change ideology to 'logDifferentRootAttributes' , but it's all opinion based!